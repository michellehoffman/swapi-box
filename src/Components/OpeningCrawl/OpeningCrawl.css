@import url('https://fonts.googleapis.com/css?family=Kanit:800,900');

* {
  margin: 0px;
}

.crawl {
  animation: scroll-up 30s linear normal;
  font-family: 'Kanit', sans-serif;
  height: 100%;
  margin-left: 15%;
  position: absolute;
  transform: translateY(800%);
  width: 70%;
  z-index: -1;
}

.crawl-text,
.film-title,
.release-date {
  font-size: 3rem;
  color: rgba(255, 232, 31);
  text-align: center;
}

.film-title {
  font-size: 5rem;
  margin: 40px 0 0 -5px;
}

.release-date {
  margin-left: -10px;
}

img {
  margin-top: 70px;
  width: 500px;
}

@keyframes scroll-up {
  0% { transform: translateY(100%); }
  100% { transform: translateY(-200%); }
}