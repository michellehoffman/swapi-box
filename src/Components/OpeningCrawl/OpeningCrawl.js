import React from 'react';
import logo from '../../images/logo.png';
import { string } from 'prop-types';
import './OpeningCrawl.css';

const OpeningCrawl = ({ title, openingCrawl, releaseDate }) => {
  return (
    <div>
      <section className="crawl">
        <article className="crawl-text">{ openingCrawl }</article>
        <p className="film-title">{ title }</p>
        <p className="release-date">Released: { releaseDate }</p>
        <img src={ logo } alt="Star Wars logo"/>
      </section>
    </div>
  )
}

OpeningCrawl.propTypes = {
  title: string.isRequired,
  openingCrawl: string,
  releaseDate: string.isRequired
}

export default OpeningCrawl;