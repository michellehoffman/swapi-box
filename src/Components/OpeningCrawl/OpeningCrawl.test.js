import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import OpeningCrawl from './OpeningCrawl';

describe('OpeningCrawl', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<OpeningCrawl />);
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should have a class name of .Crawl', () => {
    expect(wrapper.find('div').hasClass('crawl')).toEqual(true);
  });

  it('should receive props of crawl', () => {
    expect(wrapper.props().crawl).toBeDefined();
  });

  it('should render the opening crawl, film title, and release date', () => {
    expect(wrapper.find('.crawl-text').text).toEqual("It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....");
    expect(wrapper.find('.film-title').text).toEqual('A New Hope');
    expect(wrapper.find('.release-date').text).toEqual('1977-05-25');
  });

});