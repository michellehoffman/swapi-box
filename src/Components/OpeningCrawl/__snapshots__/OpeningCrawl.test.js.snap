// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpeningCrawl should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningCrawl
    characters={
        Array [
            "https://swapi.co/api/people/1/",
            "https://swapi.co/api/people/2/",
            "https://swapi.co/api/people/3/",
            "https://swapi.co/api/people/4/",
            "https://swapi.co/api/people/5/",
            "https://swapi.co/api/people/6/",
            "https://swapi.co/api/people/7/",
            "https://swapi.co/api/people/8/",
            "https://swapi.co/api/people/9/",
            "https://swapi.co/api/people/10/",
            "https://swapi.co/api/people/12/",
            "https://swapi.co/api/people/13/",
            "https://swapi.co/api/people/14/",
            "https://swapi.co/api/people/15/",
            "https://swapi.co/api/people/16/",
            "https://swapi.co/api/people/18/",
            "https://swapi.co/api/people/19/",
            "https://swapi.co/api/people/81/",
          ]
    }
    created="2014-12-10T14:23:31.880000Z"
    director="George Lucas"
    edited="2015-04-11T09:46:52.774897Z"
    episode_id={4}
    opening_crawl="It is a period of civil war.
Rebel spaceships, striking
from a hidden base, have won
their first victory against
the evil Galactic Empire.

During the battle, Rebel
spies managed to steal secret
plans to the Empire's
ultimate weapon, the DEATH
STAR, an armored space
station with enough power
to destroy an entire planet.

Pursued by the Empire's
sinister agents, Princess
Leia races home aboard her
starship, custodian of the
stolen plans that can save her
people and restore
freedom to the galaxy...."
    planets={
        Array [
            "https://swapi.co/api/planets/2/",
            "https://swapi.co/api/planets/3/",
            "https://swapi.co/api/planets/1/",
          ]
    }
    producer="Gary Kurtz, Rick McCallum"
    release_date="1977-05-25"
    species={
        Array [
            "https://swapi.co/api/species/5/",
            "https://swapi.co/api/species/3/",
            "https://swapi.co/api/species/2/",
            "https://swapi.co/api/species/1/",
            "https://swapi.co/api/species/4/",
          ]
    }
    starships={
        Array [
            "https://swapi.co/api/starships/2/",
            "https://swapi.co/api/starships/3/",
            "https://swapi.co/api/starships/5/",
            "https://swapi.co/api/starships/9/",
            "https://swapi.co/api/starships/10/",
            "https://swapi.co/api/starships/11/",
            "https://swapi.co/api/starships/12/",
            "https://swapi.co/api/starships/13/",
          ]
    }
    title="A New Hope"
    url="https://swapi.co/api/films/1/"
    vehicles={
        Array [
            "https://swapi.co/api/vehicles/4/",
            "https://swapi.co/api/vehicles/6/",
            "https://swapi.co/api/vehicles/7/",
            "https://swapi.co/api/vehicles/8/",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="crawl"
>
        <article
                className="crawl-text"
        />
        <p
                className="film-title"
        >
                A New Hope
        </p>
        <p
                className="release-date"
        >
                Released: 
        </p>
        <img
                alt="Star Wars logo"
                src="logo.png"
        />
</section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="crawl-text"
/>,
          <p
            className="film-title"
>
            A New Hope
</p>,
          <p
            className="release-date"
>
            Released: 
</p>,
          <img
            alt="Star Wars logo"
            src="logo.png"
/>,
        ],
        "className": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "crawl-text",
          },
          "ref": null,
          "rendered": null,
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A New Hope",
            "className": "film-title",
          },
          "ref": null,
          "rendered": "A New Hope",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Released: ",
              undefined,
            ],
            "className": "release-date",
          },
          "ref": null,
          "rendered": Array [
            "Released: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Star Wars logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="crawl"
>
          <article
                    className="crawl-text"
          />
          <p
                    className="film-title"
          >
                    A New Hope
          </p>
          <p
                    className="release-date"
          >
                    Released: 
          </p>
          <img
                    alt="Star Wars logo"
                    src="logo.png"
          />
</section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="crawl-text"
/>,
            <p
              className="film-title"
>
              A New Hope
</p>,
            <p
              className="release-date"
>
              Released: 
</p>,
            <img
              alt="Star Wars logo"
              src="logo.png"
/>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "crawl-text",
            },
            "ref": null,
            "rendered": null,
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A New Hope",
              "className": "film-title",
            },
            "ref": null,
            "rendered": "A New Hope",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Released: ",
                undefined,
              ],
              "className": "release-date",
            },
            "ref": null,
            "rendered": Array [
              "Released: ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Star Wars logo",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
